File: Wayne/WayneCompiler/src/Dfir_representation/Range.java
package Dfir_representation;

public class Range{
	   private int lowValue;
	   private int highValue;
	   private int singleValue;
	   private int diagramIndex;
	   private Boolean tf;
	   

	private Boolean isSingle;
	   private Boolean isBoolean;
	   
	public Range(int lowValue, int highValue, int diagramIndex){
		  this.lowValue = lowValue;
		  this.highValue = highValue;
		  this.diagramIndex = diagramIndex;
		  this.isSingle = false;
		  this.isBoolean = false;
	   }
	   public Range(int singleValue, int diagramIndex){
		      this.singleValue = singleValue;
			  this.diagramIndex = diagramIndex;    
			  this.isSingle = true;
			  this.isBoolean = false;
	   }
	   public Range(Boolean b, int diagramIndex){
			  this.tf = b;
			  this.diagramIndex = diagramIndex;    
			  this.isBoolean = true;
			  this.isSingle = false;
	   }
	   
	    public Boolean getIsBoolean() {
		    return isBoolean;
	    }
	    public void setIsBoolean(Boolean isBoolean) {
		    this.isBoolean = isBoolean;
	    }
	    public Boolean getTf() {
			return tf;
		}
		public void setTf(Boolean tf) {
			this.tf = tf;
		}
		public Boolean getIsSingle() {
			return isSingle;
		}
		public void setIsSingle(Boolean isSingle) {
			this.isSingle = isSingle;
		}
	    public int getLowValue() {
			return lowValue;
		}
		public void setLowValue(int lowValue) {
			this.lowValue = lowValue;
		}
		public int getHighValue() {
			return highValue;
		}
		public void setHighValue(int highValue) {
			this.highValue = highValue;
		}
		public int getSingleValue() {
			return singleValue;
		}
		public void setSingleValue(int singleValue) {
			this.singleValue = singleValue;
		}
		public int getDiagramIndex() {
			return diagramIndex;
		}
		public void setDiagramIndex(int diagramIndex) {
			this.diagramIndex = diagramIndex;
		}
	  
	   public void print(){
		   System.out.println("Range begin");
		   if(this.singleValue != Integer.MIN_VALUE){
			   System.out.println("SingleValue: " + this.singleValue+ ' ' + "DiagramIndex: " + this.diagramIndex);
		   } else {
			   System.out.println("LowValue: " + this.lowValue + ' ' + "HighValue: " + this.highValue + ' ' + "DiagramIndex: " + this.diagramIndex);
		   }
		   System.out.println("Range end\n");
	   }
}

File: Jasmine/JasmineCompiler/Src/DFIR_Compiler/DFIR_Compiler/Range.cs
using System;

namespace DFIR_Compiler
{
    public class Range
    {
        private int singleValue;
        private int lowValue;
        private int highValue;

        private int diagramIndex;
        private bool trueorfalse;

        private bool isSingle;
        private bool isBoolean;

        public Range(int SingleValue, int DiagramIndex)
        {
            singleValue = SingleValue;
            diagramIndex = DiagramIndex;
            isSingle = true;
            isBoolean = false;
        }

        public Range(int LowValue, int HighValue, int DiagramIndex)
        {
            lowValue = LowValue;
            highValue = HighValue;
            diagramIndex = DiagramIndex;
            isSingle = false;
            isBoolean = false;
        }

        public Range(bool booltype, int DiagramIndex)
        {
            trueorfalse = booltype;
            diagramIndex = DiagramIndex;
            isSingle = false;
            isBoolean = true;
        }

        public virtual int GetSingleValue()
        {
            return singleValue;
        }
        public virtual int GetHighValue()
        {
            return highValue;
        }
        public virtual int GetLowValue()
        {
            return lowValue;
        }
        public virtual bool GetTrueOrFalse()
        {
            return trueorfalse;
        }
        public virtual int GetDiagramIndex()
        {
            return diagramIndex;
        }
        public virtual bool GetIsSingle()
        {
            return isSingle;
        }
        public virtual bool GetIsBoolean()
        {
            return isBoolean;
        }


        public virtual void SetSingleValue(int SingleValue)
        {
            singleValue = SingleValue;
        }
        public virtual void SetLowValue(int LowValue)
        {
            lowValue = LowValue;
        }
        public virtual void SetHighValue(int HighValue)
        {
            highValue = HighValue;
        }
        public virtual void SetTrueOrFalse(bool tf)
        {
            trueorfalse = tf;
        }
        public virtual void SetDiagramIndex(int DiagramIndex)
        {
            diagramIndex = DiagramIndex;
        }
        public virtual void SetIsSingle(bool IsSingle)
        {
            isSingle = IsSingle;
        }
        public virtual void SetIsBoolean(bool IsBoolean)
        {
            isBoolean = IsBoolean;
        }


        public virtual void Print()
        {
            Console.WriteLine("Range:");
            if (singleValue != int.MinValue)
            {
                Console.WriteLine("SingleValue: " + singleValue + ' ' + "DiagramIndex: " + diagramIndex);
            }
            else
            {
                Console.WriteLine("LowValue: " + lowValue + ' ' + "HighValue: " + highValue + ' ' + "DiagramIndex: " + diagramIndex);
            }
        }

    }
}
